### ISSUE TITLE
"Show in Organizer" button does not URL-encode filter string, so it can become corrupt

### ISSUE THREAD

**Issue by DHager** [DestinyItemManager/DIM#7167]
### DIM Version

Version 6.86.0 (release), built on 10/10/2021, 7:08:06 PM

### Browser Details

Chrome 94.0.4606.81

### OS Details

Windows 7

#### TLDR diagnosis & fix

The link generated for the "Show in Organizer" button does not url-encode the filter string. Among other possible problems, this means that anything after a hash mark (`#`) is discarded when the Organizer view is loaded. It looks like the problem code is in `src/app/compare/selectors.ts:compareOrganizerLinkSelector()`, which probably needs to have `encodeURIComponent(session.query)`. 

I'd make a PR myself, but unfortunately my local dev setup is a little messed up at the moment after changing jobs.

#### Repro steps

Suppose I'm looking at some weapons I have used the notes-field to "custom tag" for PVP:

```
is:handcannon notes:#pvp stat:rpm:=140
```

I use this search and then click the "Compare Items" button, bringing up the horizontal listing across the bottom of the screen. Then I decide I want to view them differently, perhaps to see the full Notes field, so I click on the "Show In Organizer" button in the top-right of the drawer.

This takes me to the organizer view, but the wrong items are showing! My filter-string is now:

```
is:handcannon notes:
```

**Expected:** Filter string should not change from what was seen in the Inventory view.

**Comment by bhollis** [comment]
Thanks! It's not a great substitute but if you type `.` in GitHub it'll open a pretty nice online editor. You can't run DIM in it though.
